pkg/dockerregistry/server/blobdescriptorservice.go

```diff
@@ -25,6 +38,137 @@ type blobDescriptorService struct {
 	distribution.BlobDescriptorService
 }
 
+// Stat returns a a blob descriptor if the given blob is either linked in repository or is referenced in
+// corresponding image stream. This method is invoked from inside of upstream's linkedBlobStore. It expects
+// a proper repository object to be set on given context by upper openshift middleware wrappers.
 func (bs *blobDescriptorService) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
-	return dockerRegistry.BlobStatter().Stat(ctx, dgst)
+	// if there is a repo layer link, return its desc
+	desc, err := bs.BlobDescriptorService.Stat(ctx, dgst)
```

Comment:
> So the *only* place in memory cache was being used was here?  No one else is expecting to make blob store calls that are inexpensive (because they're guarded by the cache) but will only find out that they are very expensive?

I made a bad mistake. The inmemory cache is used also for stat-ing the global blob store. If turned off, all the stats will take much longer when done against slow storage.

And I was wrong also in the case of the order. Our `blobDescriptorService` actually wraps the `cachedBlobStatter` (see the updated comment above). So functional-wise we're fine having it enabled. Unfortunately, it doubles the memory usage per blob cached. We'd better push upstream an ability for defining our own `blobDescriptorCacheProvider`.

That can be easily done as a follow-up. For now, I'll just re-enable the inmemory cache.